### PR TITLE
Bump targetSdk & minSdk

### PR DESCRIPTION
**Pull request by Goooler**
- Bump targetSdk to 31
- Elevate minSdk to 21 (sdk 19 devices [less than 2%](https://www.appbrain.com/stats/top-android-sdk-versions), no need to compat)

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -53,10 +53,8 @@ android {
   compileSdkVersion 31
 
   defaultConfig {
-    // 19+ required by the webview_flutter plugin
-    minSdkVersion 19
-    targetSdkVersion 30
-    multiDexEnabled true
+    minSdkVersion 21
```
Comment (robertodoering) [inline android/app/build.gradle:56]: Is there a reason to bump the min sdk?
Less than 1% of users (~20) are on android 4.4 (api 19), so it's not a big deal but if we don't have any benefit from it we should probably leave it as is.

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -20,6 +20,7 @@
     <activity
       android:name=".MainActivity"
       android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
+      android:exported="true"
```
Comment (robertodoering) [inline android/app/src/main/AndroidManifest.xml:23]: What's the reason for adding this?

Comment (Goooler) [reply android/app/src/main/AndroidManifest.xml:23]: Resolve conflict when merging manifest files.

File: android/app/build.gradle (hunk 1 above)
Comment (Goooler) [reply android/app/build.gradle:56]: Less compatible issues, less dependencies depends on.

File: android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (robertodoering) [reply android/app/src/main/AndroidManifest.xml:23]: Ah I see, this became mandatory after bumping the target sdk

File: android/app/build.gradle (hunk 1 above)
Comment (robertodoering) [reply android/app/build.gradle:56]: That's true, I will test it when I have the time to make sure nothing broke

Comment (Goooler) [reply android/app/build.gradle:56]: We use `minSdk = 23` & `targetSdk = 31` for days, it should be same as sdk 30 in perssmions and files.

**Review by robertodoering — APPROVED** [review]
Looks good! Thanks

**Pull request merged**